doc/architecture/adr/008_foreign_keys.md

```diff
@@ -0,0 +1,32 @@
+# ADR-008 Einsatz von foreign keys auf der DB
+
+Status: **Vorschlag**
+
+Entscheid: **Offen**
+
+## Kontext
+
+Referentielle Integrität wird primär über Applikations Code (`dependent: :destroy`, `protect_if`)
+sichergestellt. Vereinzelt sind auch foreign key constraints auf Tabellen definiert, die dann beim
+speichern eine `InvalidForeignKey` Exception werfen, welche gefangen und behandelt wird.
+
+## Optionen
```

Comment:
### Foreign Keys, aber nur als DB-Hilfsmittel

Foreign Keys als DB-Constraint finde ich noch genauso sinnvoll wie Unique Indices. Sie sind ein Hilfsmittel und unterstützen die Konsistenz, sollten aber primär in der DB bleiben. Da sie aber in der Applikationlogik nicht unbedingt sichtbar sind, müssen die entsprechenden Framework-Möglichkeiten genutzt werden. Damit sind Foreignkeys lediglich ein weiteres Sicherheitsnetz, dass in genutzt werden kann, aber nicht muss. 

Für FKs spricht, dass man meist ohnehin Indices auf den Fremdschlüsseln möchte, da diese für JOINs verwendet werden. Bei wichtigen Verbindungen kann dies dann auch noch mit einem constraint erweitert werden, der die Integrität sicherstellt.